frontend/src/lib/components/CommandBar/SearchResult.tsx

```diff
@@ -86,9 +92,23 @@ type ResultNameProps = {
 }
 
 export const ResultName = ({ result }: ResultNameProps): JSX.Element | null => {
+    const { aggregationLabel } = useValues(groupsModel)
+    const { cohortsById } = useValues(cohortsModel)
+    const { mathDefinitions } = useValues(mathsLogic)
+
     const { type, extra_fields } = result
     if (type === 'insight') {
-        return extra_fields.name ? <span>{extra_fields.name}</span> : <i>{extra_fields.derived_name}</i>
+        return extra_fields.name ? (
+            <span>{extra_fields.name}</span>
+        ) : (
+            <i>
+                {summarizeInsight(extra_fields.query as Node | null, extra_fields.filters as Partial<FilterType>, {
+                    aggregationLabel,
+                    cohortsById,
+                    mathDefinitions,
+                })}
+            </i>
+        )
```

Comment:
This doesn't seem right, because there shouldn't _be_ insights that have neither `name` nor `derived_name`. Facilitating searchability is a major reason for `derived_name` being saved, so we shouldn't paper over this. Should we revert this and fix the root cause, or am I missing something here? @thmsobrmlr

Reply:
Hmm, I've reimplemented what we do for saved insights as `derived_name` isn't set for insights in production. Happy to look into it, but would appreciate pointers on how/where we set it. And wouldn't revert this change for now, as we do the same thing in other places and it works for now.